fathom/src/surface/elaboration.rs

```diff
@@ -1900,118 +1990,284 @@ impl<'interner, 'arena, 'error> Context<'interner, 'arena, 'error> {
         let operand_types = Option::zip(lhs_type.match_prim_spine(), rhs_type.match_prim_spine());
 
         let (fun, output_type) = match (op, operand_types) {
```

Comment:
It’s a bit of a shame that the formatting went out the window on the operator matches… really wish `rustfmt` let you add annotations to expressions for stuff like this 🤔

Won't hold off the PR for this but might have a look myself at some stage.